src/allmydata/test/common_web.py

```diff
@@ -36,14 +37,22 @@
     TahoeLAFSRequest,
 )
 
+
+class VerboseError(Error):
+    """Include the HTTP body response too."""
+
+    def __str__(self):
+        return Error.__str__(self) + " " + ensure_str(self.response)
+
+
 @inlineCallbacks
 def do_http(method, url, **kwargs):
     response = yield treq.request(method, url, persistent=False, **kwargs)
     body = yield treq.content(response)
     # TODO: replace this with response.fail_for_status when
     # https://github.com/twisted/treq/pull/159 has landed
     if 400 <= response.code < 600:
-        raise Error(response.code, response=body)
+        raise VerboseError(response.code, response=body)
```

Comment:
I wonder what a test failure that ends up hitting this exception looks like.  A ten page html/css smear?  I like the idea of more informative failures but cringe a bit at the thought of reading an html-formatted traceback in my test failures.

Reply:
It leaves out the body altogether, which makes it quite difficult to debug integration tests, because the server-side exception is hidden in the depths of foolscap.